Topics/Development_Process/WSL.md

```diff
@@ -0,0 +1,113 @@
+
+    wsl --shutdown
+
+To delete a distribution, run:
+
+    wsl --unregister <distribution>
+
+### WSL Help, Update, and Version
+
+To update WSL, run:
+
+    wsl --update
+
+To see WSL version:
+
+    wsl --version
+
+To see possible commands with WSL, run:
+
+    wsl --help
+
+### Distribution Specific Configuration Files
+
+"/etc/wsl.conf" is the configuration file read by WSL in startup when booting the distribution, if it exists. A simple search in the documentation for available configuration options can help automate things when you are launching WSL ! Note that more features have been added in newer builds, so version is important to take care of, which is available [here](https://learn.microsoft.com/en-us/windows/wsl/release-notes).
+
+----
+
+## WSL 1 Support
+
+Microsoft changed the way they parse Windows Path files between WSL 1 and WSL 2, and it created a deadlock where trying to run Windows Docker Desktop based on WSL 2 will lead to Path errors, and Windows Docker Desktop cannot be run on WSL 1 due different virtualization capabilities. There is no visible solution to the author's best knowledge and extensive research to bypass setup scripts involved. It is recommended to use Docker only on Linux instead, especially for the sake of best compatibility (Docker would still work in a distribution running on WSL). Thus, we are going to discuss how to run WSL 1 for any similar reasons that may require it.
+
+Following the installation steps in this document, it is assumed by WSL that the user wants to use the latest version, version 2. To change this default setting, run:
+
+    wsl --set-default-version 1
+
+To change the version of a specific distribution instead, run:
+
+    wsl --set-version <distribution> <version>
+
+This allows backwards compatibility. To change WSL 1 distribution launch settings:
+
+1) Run regedit.exe
+2) Navigate to HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+3) Use [WSL_DISTRIBUTION_FLAGS](https://learn.microsoft.com/en-us/windows/win32/api/wslapi/ne-wslapi-wsl_distribution_flags) to change launch settings for distributions. For example, 0x7 implies all flags are enabled (7 = 1 + 2 + 4).
+
+**NOTE:** Modifying registers can potentially damage and crash you system and data. It is important to be responsible and think twice before implementing changes.
+
+This information should be mostly sufficient, and further information is available on official documentation.
+
+----
+
+## Terminology
+- [**Virtualization**](https://www.ibm.com/topics/virtualization): the action of creating an abstraction layer over computer hardware that allows the hardware elements of a single computer like processors, memory, storage and more to be divided into multiple virtual computers, commonly called virtual machines (VMs). This action allows you to run Linux on top of Windows without shutting it down!
+
+- **Distribution**: The operating system being virtualized on your Windows machine, usually some flavor of Linux.
```

Comment:
Just wondering if theres any common issues with WSL, just because many times when I read documentation and try it out, something doesn't work correctly. It's nice when a doc has a section saying "common issues" with their solutions.

Reply:
I added Other Notes section to cover basic approach to problems since following the discussed steps to setup WSL would work almost all time, given their complete and considerate approach. The community is rich with solutions for other problems as well.